WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt

```diff
@@ -199,16 +229,54 @@ class ReaderPostDetailViewModel @Inject constructor(
         )
     }
 
+    private fun updateFollowButtonUiState(
+        currentUiState: ReaderPostDetailsUiState,
+        isFollowed: Boolean
+    ) {
+        val updatedFollowButtonUiState = currentUiState
+                .headerUiState
+                .followButtonUiState
+                .copy(isFollowed = isFollowed)
+
+        val updatedHeaderUiState = currentUiState
+                .headerUiState
+                .copy(followButtonUiState = updatedFollowButtonUiState)
+
+        _uiState.value = currentUiState.copy(headerUiState = updatedHeaderUiState)
+    }
+
+    private fun updatePostActions(post: ReaderPost) {
+        _uiState.value = _uiState.value?.copy(
+                actions = postDetailUiStateBuilder.buildPostActions(
+                        post,
+                        this@ReaderPostDetailViewModel::onButtonClicked
+                )
+        )
+    }
+
+    private fun updateRelatedPostsUiState(state: FetchRelatedPostsState.Success) {
+        _uiState.value = _uiState.value?.copy(
+                localRelatedPosts = RelatedPosts(posts = state.localRelatedPosts, isGlobal = false),
+                globalRelatedPosts = RelatedPosts(posts = state.globalRelatedPosts, isGlobal = true)
+        )
+    }
     data class ReaderPostDetailsUiState(
         val postId: Long,
         val blogId: Long,
         val headerUiState: ReaderPostDetailsHeaderUiState,
         val moreMenuItems: List<SecondaryAction>? = null,
-        val actions: ReaderPostActions
-    )
+        val actions: ReaderPostActions,
+        /* related posts from the same site as the current post */
```

Comment:
Suggestion (💡): Maybe these comments could be located closer to the source? Like `ReaderPostActions.handleRelatedPostsResponse(...)`?

Reply:
Comment already exists [in the source](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java#L430).
`// split into posts from the passed site (local) and from across wp.com (global)`

It looks sufficient to me. Wdyt?

Reply:
Yeap, definitely, to me too. Why I comment on that is because maybe we should consider removing those from here. Wdyt? Do we really need them there?

Reply:
Got it, comment removed in d54dfd2.

Reply:
Many thanks once more @ashiagr !

Feel free to merge this PR whenever you are ready. 🌟